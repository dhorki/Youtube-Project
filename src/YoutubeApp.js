import React, { useReducer } from 'react';
import { EnvironmentContext } from './contexts/EnvironmentContext';
import { AppRouter } from './routers/AppRouter';
import { styles } from './constants/constants';
import { appReducer } from './reducers/appReducer';

export const YoutubeApp = () => {
  const init = {
    theme: styles.colors.light,
    sidebarShow: false,
    q: '',
  };

  const [environment, dispatchEnv] = useReducer(appReducer, init);

  return (
    <EnvironmentContext.Provider value={{ environment, dispatchEnv }}>
      <AppRouter />
    </EnvironmentContext.Provider>
  );
};
