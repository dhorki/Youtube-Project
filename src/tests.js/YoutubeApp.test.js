import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { YoutubeApp } from '../YoutubeApp';

describe('YoutubeApp', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<YoutubeApp />);
    expect(wrapper).toMatchSnapshot();
  });
});
