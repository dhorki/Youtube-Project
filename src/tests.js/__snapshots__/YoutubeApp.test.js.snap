// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YoutubeApp should render correctly 1`] = `
<ContextProvider
  value={
    Object {
      "dispatchEnv": [Function],
      "environment": Object {
        "q": "",
        "sidebarShow": false,
        "theme": Object {
          "appIcon": "#ff0200",
          "background": "#F9F9F9",
          "barBackground": "#ffffff",
          "inputBackground": "#ffffff",
          "inputBorder": "#cccccc",
          "linkActive": "#ff0200",
          "linkHoverBackground": "#ececec",
          "primary": "#030303",
          "searchButtonBackground": "#f8f8f8",
          "secondary": "#606060",
        },
      },
    }
  }
>
  <AppRouter />
</ContextProvider>
`;
